catalog/security/catalog-security-policyplugin/src/main/java/org/codice/ddf/catalog/security/CatalogPolicy.java

```diff
@@ -0,0 +1,274 @@
+        }
+        return null;
+    }
+
+    /**
+     * Get the KeyValuePermission that have been parsed from the permissions string
+     *
+     * @return
+     */
+    public Map<String, Set<String>> getCreatePermissionMap() {
+        return Collections.unmodifiableMap(createPermissionMap);
+    }
+
+    /**
+     * Get the KeyValuePermission that have been parsed from the permissions string
+     *
+     * @return
+     */
+    public Map<String, Set<String>> getUpdatePermissionMap() {
+        return Collections.unmodifiableMap(updatePermissionMap);
+    }
+
+    /**
+     * Get the KeyValuePermission that have been parsed from the permissions string
+     *
+     * @return
+     */
+    public Map<String, Set<String>> getDeletePermissionMap() {
+        return Collections.unmodifiableMap(deletePermissionMap);
+    }
+
+    /**
+     * Get the KeyValuePermission that have been parsed from the permissions string
+     *
+     * @return
+     */
+    public Map<String, Set<String>> getReadPermissionMap() {
+        return Collections.unmodifiableMap(readPermissionMap);
+    }
+
+    /**
+     * Setter used by the ui to set the permissions/attributes
+     *
+     * @param permStrings
+     */
+    public void setCreatePermissions(String[] permStrings) {
+        if (permStrings != null) {
+            this.createPermissions = Arrays.copyOf(permStrings, permStrings.length);
+            parsePermissionsFromString(permStrings, createPermissionMap);
+        }
+    }
+
+    /**
+     * Setter used by the ui to set the permissions/attributes
+     *
+     * @param permStrings
+     */
+    public void setUpdatePermissions(String[] permStrings) {
+        if (permStrings != null) {
+            this.updatePermissions = Arrays.copyOf(permStrings, permStrings.length);
+            parsePermissionsFromString(permStrings, updatePermissionMap);
+        }
+    }
+
+    /**
+     * Setter used by the ui to set the permissions/attributes
+     *
+     * @param permStrings
+     */
+    public void setDeletePermissions(String[] permStrings) {
+        if (permStrings != null) {
+            this.deletePermissions = Arrays.copyOf(permStrings, permStrings.length);
+            parsePermissionsFromString(permStrings, deletePermissionMap);
+        }
+    }
+
+    /**
+     * Setter used by the ui to set the permissions/attributes
+     *
+     * @param permStrings
+     */
+    public void setReadPermissions(String[] permStrings) {
+        if (permStrings != null) {
+            this.readPermissions = Arrays.copyOf(permStrings, permStrings.length);
+            parsePermissionsFromString(permStrings, readPermissionMap);
+        }
+    }
+
+    /**
+     * Parses a string  array representation of permission attributes
+     *
+     * @param permStrings String array of permissions to parse
+     */
+    private void parsePermissionsFromString(String[] permStrings, Map<String, Set<String>> permissions) {
+        permissions.clear();
```

Comment:
I'm a little worried about concurrent access to the maps and wonder if we shouldn't use a stack-local map for the processing and then do a final swap in a synchronized block and then synch the methods that get the `unmodifiableMap`s of them.